test/extended/deployments/deployments.go

```diff
@@ -171,12 +189,10 @@ var _ = g.Describe("deploymentconfigs", func() {
 				secondDeploymentName := deployutil.DeploymentNameForConfigVersion(name, 2)
 				secondDeployerExists := len(deploymentNamesToDeployers[secondDeploymentName]) == 1
 
-				return hasCorrectLatestVersion &&
-					secondDeploymentExists &&
-					firstDeployerRemoved && secondDeployerExists, nil
+				return secondDeploymentExists
 			}
-
 			err = wait.PollImmediate(500*time.Millisecond, 10*time.Second, condition)
+			printDebugInfo(err, oc, name)
```

Comment:
Instead of needing printDebugInfo, why not have the condition include this info in the error?  Or have the failure trap print it out?  both of those are more readable and easier to reuse.